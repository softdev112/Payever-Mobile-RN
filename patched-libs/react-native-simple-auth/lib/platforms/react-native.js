import { Linking } from 'react-native'; // eslint-disable-line import/no-unresolved, max-len

export const dance = authUrl => Linking.openURL(authUrl)
  .then(() => new Promise((resolve, reject) => {
    const handleUrl = (url) => {
      if (!url || url.indexOf('fail') > -1) {
        reject(url);
      } else {
        resolve(url);
      }
    };

    const onLinkChange = ({ url }) => {
      Linking.removeEventListener('url', onLinkChange);
      handleUrl(url);
    };

    Linking.addEventListener('url', onLinkChange);
  }));

export const request = fetch;
