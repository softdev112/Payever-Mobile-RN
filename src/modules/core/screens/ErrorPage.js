import { Component } from 'react';
import type { Navigator } from 'react-native-navigation';
import { Button, ErrorBox, Header, StyleSheet, View } from 'ui';

export default class ErrorPage extends Component {
  static navigatorStyle = {
    navBarHidden: true,
  };

  props: {
    message: string;
    navigator: Navigator;
    onBack?: () => {};
  };

  onGoBack() {
    const { navigator, onBack } = this.props;

    if (onBack) {
      onBack();
    } else {
      navigator.pop({ animated: true });
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Header>Error!</Header>
        <View style={styles.errorContainer}>
          <ErrorBox message={this.props.message} />
          <Button
            style={styles.button}
            title={'Back'}
            onPress={::this.onGoBack}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  errorContainer: {
    alignItems: 'center',
    paddingTop: '15%',
    flex: 1,
  },

  error: {
    width: '80%',
  },

  button: {
    marginTop: 30,
    width: 120,
    elevation: 2,
  },
});