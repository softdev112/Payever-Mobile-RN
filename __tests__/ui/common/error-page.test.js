import 'react-native';
import renderer from 'react-test-renderer';

import ErrorPage from '../../../src/modules/core/screens/ErrorPage';

describe('src/modules/core/screens/ErrorPage.js', () => {
  it('src/modules/core/screens/ErrorPage.js renders correctly', () => {
    const tree = renderer.create(
      <ErrorPage message="Hello World!" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});