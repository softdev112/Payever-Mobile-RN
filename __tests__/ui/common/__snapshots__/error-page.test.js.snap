// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/modules/core/screens/ErrorPage.js src/modules/core/screens/ErrorPage.js renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#e1e1e1",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 10,
          "padding": 20,
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Helvetica Neue",
          },
          Object {
            "color": "#3d3d3d",
            "textAlign": "center",
          },
        ]
      }>
      Error
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      duration={0}
      ellipsizeMode="tail"
      message="Hello World!"
      style={
        Array [
          Object {
            "borderColor": "red",
            "borderRadius": 4,
            "borderWidth": 1,
            "color": "red",
            "padding": 5,
            "textAlign": "center",
          },
          Object {
            "width": 600,
          },
        ]
      }>
      Hello World!
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0084ff",
          "borderRadius": 4,
          "marginTop": 10,
          "opacity": 1,
          "padding": 3,
          "width": 120,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontWeight": "500",
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }>
        Go To Login
      </Text>
    </View>
  </View>
</View>
`;
